ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -0,0 +1,69 @@
+import requests
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+
+_GCP_METADATA_URI = "http://metadata.google.internal/computeMetadata/v1/"
+_GCP_METADATA_URI_HEADER = {"Metadata-Flavor": "Google"}
+
+# GCE common attributes
+# See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+_GCE_ATTRIBUTES = {
+    "project_id": "project/project-id",
+    "instance_id": "instance/id",
+    "zone": "instance/zone",
+}
+
+_ATTRIBUTE_URI_TRANSFORMATIONS = {
+    "instance/zone": lambda v: v.split("/")[-1] if "/" in v else v
+}
+
+_GCP_METADATA_MAP = {}
```

Comment:
Python coding style question: This is an internal cache. Should we use all-uppercase variable names for non-constants?

Reply:
IIRC we generally do, but probably not consistently. `pylint` will complain without an exemption, so I think we generally uppercase instead of `#noqa`.

Reply:
This is moved to a class field. I think even having a non constant global isn't great in the first place.